### PR TITLE
docs: consistent @see jsdoc tags

### LINKED ISSUE #2057: Unify @see jsdoc tags
While working on some @see tags, because @ST-DDT and @xDivisionByZerox pointed that to me, I found that your current code base is very unstructurized in terms of the @see tag.

Sometimes sentences are starting with an upper case, sometimes with a lower case.
Sometimes they end or end not with a period (`.`).
Sometimes they have no description at all.

### PR DESCRIPTION
**Pull request by ST-DDT**
Fixes #2057

- #2057

Requires that all `@see` tags that point to faker methods follow the following pattern:

````txt
@see faker.<module>.<method>(.*): [A-Z]___.
````

---

All changes are in a single PR to make it easier to cross compare the individual changes.
Feedback and re-phrasing suggestions welcome.

### CODE REVIEW

File: src/definitions/finance.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export type FinanceDefinition = LocaleEntry<{
    * The pattern by (lowercase) issuer name used to generate credit card codes.
    * `L` will be replaced by the check bit.
    *
-   * @see faker.helpers.replaceCreditCardSymbols()
+   * @see faker.helpers.replaceCreditCardSymbols(): For more information about the how the pattern is used.
```
Comment (xDivisionByZerox) [inline src/definitions/finance.ts]:
```suggestion
   * @see faker.helpers.replaceCreditCardSymbols(): For more information about how the pattern is used.
```

File: src/definitions/phone_number.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export type PhoneNumberDefinition = LocaleEntry<{
    * `!` will be replaced by a random digit (2-9).
    * (e.g. `!##-!##-####` -> 272-285-0453)
    *
-   * @see faker.helpers.replaceSymbolWithNumber(format)
+   * @see faker.helpers.replaceSymbolWithNumber(format): For more information about the how the patterns are used.
```
Comment (xDivisionByZerox) [inline src/definitions/phone_number.ts]:
```suggestion
   * @see faker.helpers.replaceSymbolWithNumber(format): For more information about how the patterns are used.
```

File: src/modules/datatype/index.ts [hunk 3]
```diff
@@ -158,8 +158,8 @@ export class DatatypeModule {
    *    When not provided or larger than `8640000000000000`, `2100-01-01` is considered
    *    as maximum generated date. Defaults to `4102444800000`.
    *
-   * @see faker.date.anytime()
-   * @see faker.date.between()
+   * @see faker.date.anytime(): For the new method when generating a random date in either the past or future.
+   * @see faker.date.between(): For the new method when generating a random date between the two dates.
```
Comment (xDivisionByZerox) [inline src/modules/datatype/index.ts]:
```suggestion
   * @see faker.date.between(): For the new method when generating a random date in between two dates.
```

File: src/modules/image/index.ts [hunk 4]
```diff
@@ -508,8 +508,8 @@ export class ImageModule {
    * @param height The height of the image. Defaults to `480`.
    * @param randomize Whether to randomize the image or not. Defaults to `false`.
    *
-   * @see faker.image.url()
-   * @see faker.image.urlLoremFlickr()
+   * @see faker.image.url(): To generate a random image url (has less options, but doesn't break when LoremFlickr is removed).
```
Comment (xDivisionByZerox) [inline src/modules/image/index.ts]: I'm not sure abou the "doesn't break when LoremFlickr is removed". This seems oddly specific. Just leaving the comment and comming back later with a suggestion.

File: src/modules/person/index.ts [hunk 5]
```diff
@@ -268,7 +269,10 @@ export class PersonModule {
   }
 
   /**
-   * Returns a random sex type.
+   * Returns a random sex type (usable in parameters).
```
Comment (xDivisionByZerox) [inline src/modules/person/index.ts]: Not sure about the parenthesis. Maybe just start a new sentence explaining the usage.

File: src/modules/random/index.ts [hunk 6]
```diff
@@ -23,8 +23,8 @@ export class RandomModule {
   /**
    * Returns a random word.
    *
-   * @see faker.lorem.word()
-   * @see faker.word.sample()
+   * @see faker.lorem.word(): To generate a random placeholder/unreadable word.
```
Comment (xDivisionByZerox) [inline src/modules/random/index.ts]: I mean it is very readable if you know how to speak latin.

```suggestion
   * @see faker.lorem.word(): To generate a random placeholder word.
```

File: src/modules/random/index.ts [hunk 7]
```diff
@@ -162,8 +162,8 @@ export class RandomModule {
    * @param count.min The minimum number of words. Defaults to `1`.
    * @param count.max The maximum number of words. Defaults to `3`.
    *
-   * @see faker.lorem.words()
-   * @see faker.word.words()
+   * @see faker.lorem.words(): To generate a sequence of random placeholder/unreadable words.
```
Comment (xDivisionByZerox) [inline src/modules/random/index.ts]:
```suggestion
   * @see faker.lorem.words(): To generate a sequence of random placeholder words.
```

File: test/scripts/apidoc/verify-jsdoc-tags.spec.ts [hunk 8]
```diff
@@ -201,6 +201,14 @@ describe('verify JSDoc tags', () => {
...
                   ')'
                 );
+                expect(
+                  link,
+                  "Expect method reference to have a ': ' after the parenthesis"
+                ).toContain('): ');
+                expect(
+                  link,
+                  'Expect method reference to have a description sentence ending with a dot'
+                ).toMatch(/\): [A-Z].{9,}\.$/);
```
Comment (xDivisionByZerox) [inline test/scripts/apidoc/verify-jsdoc-tags.spec.ts]: The length of 9 to represent a sentence is choosen randomly or based on some research you have done?

**Review by xDivisionByZerox — CHANGES_REQUESTED** [review]

File: test/scripts/apidoc/verify-jsdoc-tags.spec.ts (hunk 8 above)
Comment (ST-DDT) [reply test/scripts/apidoc/verify-jsdoc-tags.spec.ts]: It is chosen randomly => Supposedly more than 1 word.

File: src/modules/person/index.ts (hunk 5 above)
Comment (ST-DDT) [reply src/modules/person/index.ts]: Replaced.

File: test/scripts/apidoc/verify-jsdoc-tags.spec.ts (hunk 8 above)
Comment (ST-DDT) [reply test/scripts/apidoc/verify-jsdoc-tags.spec.ts]: I could change it to 8+capital letter dot=10 or just 10 to make it less arbitrary.

File: src/modules/image/index.ts (hunk 4 above)
Comment (matthewmayer) [reply src/modules/image/index.ts]: "less options" is also bad grammar, it should be "fewer options"

**Review by matthewmayer — CHANGES_REQUESTED** [review]
There's quite a lot of inconsistency in the phrasing like:

* If you wish to generate foo
* If you would like to generate foo
* For a method that returns foo
* To generate foo

File: src/modules/image/index.ts (hunk 4 above)
Comment (matthewmayer) [reply src/modules/image/index.ts]: How about

To generate a random image url (has fewer options, uses multiple image providers).

File: test/scripts/apidoc/verify-jsdoc-tags.spec.ts (hunk 8 above)
Comment (xDivisionByZerox) [reply test/scripts/apidoc/verify-jsdoc-tags.spec.ts]: Could you split these expectation into two statements? Basically:

- Expect method reference to have a description ending with a dot
- Expect method reference to have a description to be a sentence

**Comment by ST-DDT** [comment]
Here are all `@see` tags with plain text:

<details>
<summary>`@see` Tags</summary>

````txt
@see faker.company.name(): For a method that returns a complete company name.
@see faker.date.anytime(): For the new method when generating a random date in either the past or future.
@see faker.date.between(): For dates in a specific range.
@see faker.date.between(): For the new method when generating a random date in between two dates.
@see faker.date.future(): For dates explicitly in the future.
@see faker.date.future(): For dates further in the future (years instead of days).
@see faker.date.past(): For dates explicitly in the past.
@see faker.date.past(): For dates further back in time (years instead of days).
@see faker.date.recent(): For dates in the recent past (days instead of years).
@see faker.date.soon(): For dates in the near future (days instead of years).
@see faker.finance.accountNumber(): For the new method that replaces this one.
@see faker.finance.maskedNumber(): For the new method that replaces this one.
@see faker.helpers.fromRegExp(): Generates a string matching the given regex like expressions.
@see faker.helpers.mustache(): For more information about how the phrases are generated.
@see faker.helpers.mustache(): If you wish to use custom functions for resolution.
@see faker.helpers.replaceCreditCardSymbols(): For more information about how the pattern is used.
@see faker.helpers.replaceSymbolWithNumber(format): For more information about how the patterns are used.
@see faker.helpers.replaceSymbols(): For more information about how the pattern is used.
@see faker.image.url(): To generate a random image url (has less options, but doesn't break when LoremFlickr is removed).
@see faker.image.urlLoremFlickr(): To generate a random image url from LoremFlickr.
@see faker.internet.displayName(): If you wish to generate a Unicode display name.
@see faker.internet.userName(): If you wish to generate a plain ASCII username.
@see faker.location.city(): For the new method that replaces this one.
@see faker.location.street(): For the new method that replaces this one.
@see faker.location.zipCode(): For the new method that replaces this one.
@see faker.lorem.word(): To generate a random placeholder word.
@see faker.lorem.words(): To generate a sequence of random placeholder words.
@see faker.number.bigInt(): For the new method when generating a random bigint.
@see faker.number.binary(): If you would like to generate a `binary number` (within a range).
@see faker.number.float(): For the new method when generating a random floating-point number.
@see faker.number.hex(): For the new method when generating a random hexadecimal number.
@see faker.number.int(): For the new method when generating a random integer.
@see faker.number.int(): If you would like to generate a `number` (within a range).
@see faker.number.octal(): If you would like to generate an `octal number` (within a range).
@see faker.person.gender(): If you would like to generate gender related values.
@see faker.person.gender(): If you would like to use the gender value in forms.
@see faker.person.sex(): If you would like to generate binary-gender values.
@see faker.person.sex(): If you would like to use the sex value in forms.
@see faker.person.sexType(): If you would like to use the sex value as a parameter.
@see faker.seed(): If you wish to generate reproducible values.
@see faker.setDefaultRefDate(): If you wish to generate reproducible dates.
@see faker.setDefaultRefDate(): If you wish to generate reproducible relative dates.
@see faker.string.alpha(): For the new method replacing this one.
@see faker.string.alphanumeric(): For the new method replacing this one.
@see faker.string.binary(): If you would like to generate a `binary string` with a given length (range).
@see faker.string.hexadecimal(): For the new method when generating a random hexadecimal string.
@see faker.string.numeric(): For the new method replacing this one.
@see faker.string.numeric(): If you would like to generate a `string` of digits with a given length (range).
@see faker.string.octal(): If you would like to generate an `octal string` with a given length (range).
@see faker.string.sample(): For the new method when generating a random string.
@see faker.string.uuid(): For the new method when generating a random UUID.
@see faker.word.sample(): To generate a random existing word.
@see faker.word.words(): To generate a sequence of random existing words.
@see mergeLocales() For more information about how the locales are merged.
````

</details>

Any suggestion on how to simplify them to a few patterns?

We probably need at least two:

- deprecated method replacements
- alternative method suggestions

File: test/scripts/apidoc/verify-jsdoc-tags.spec.ts (hunk 8 above)
Comment (ST-DDT) [reply test/scripts/apidoc/verify-jsdoc-tags.spec.ts]: Changed

File: src/modules/image/index.ts (hunk 4 above)
Comment (ST-DDT) [reply src/modules/image/index.ts]: Changed

**Comment by matthewmayer** [comment]
As a first step we can replace "If you would like to generate" and "If you wish to generate" and "To generate" with 

"For generating"

**Comment by matthewmayer** [comment]
If its an identical replacement method (just a rename, no parameter/behavior changes) i think we could standardise on "For the replacement method"

File: test/scripts/apidoc/verify-jsdoc-tags.spec.ts [hunk 9]
```diff
@@ -201,6 +201,22 @@ describe('verify JSDoc tags', () => {
...
+                  "Expect method reference to have a ': ' after the parenthesis"
+                ).toContain('): ');
+                expect(
+                  link,
+                  'Expect method reference to have a description starting with a capital letter'
+                ).toMatch(/\): [A-Z]/);
+                expect(
+                  link,
+                  'Expect method reference to have a description sentence'
+                ).toMatch(/\): .{10,}$/);
```
Comment (ST-DDT) [inline test/scripts/apidoc/verify-jsdoc-tags.spec.ts]: Should we change it like this:

```suggestion
                expect(
                  link,
                  'Expect method reference to start with a standard description phrase'
                ).toMatch(
                  /\): (?:For generating |For more information about |For using |For the replacement method)/
                );
```

**Comment by ST-DDT** [comment]
Ready for review.

File: src/modules/datatype/index.ts [hunk 10]
```diff
@@ -22,8 +22,8 @@ export class DatatypeModule extends SimpleModuleBase {
    *
    * @throws When options define `max < min`.
    *
-   * @see faker.number.int() for the default precision of `1`
-   * @see faker.number.float() for a custom precision
+   * @see faker.number.int(): For generating a random integer in the future.
```
Comment (matthewmayer) [inline src/modules/datatype/index.ts]: i dont think the "in the future" is necessary

File: src/modules/datatype/index.ts [hunk 11]
```diff
@@ -22,8 +22,8 @@ export class DatatypeModule extends SimpleModuleBase {
...
+   * @see faker.number.float(): For generating a random floating-point number in the future.
```
Comment (matthewmayer) [inline src/modules/datatype/index.ts]: i dont think the "in the future" is necessary

File: src/modules/datatype/index.ts [hunk 12]
```diff
@@ -334,7 +334,8 @@ export class DatatypeModule extends SimpleModuleBase {
    * @param options.prefix Prefix for the generated number. Defaults to `'0x'`.
    * @param options.case Case of the generated number. Defaults to `'mixed'`.
    *
-   * @see faker.string.hexadecimal()
+   * @see faker.string.hexadecimal(): For generating a random hexadecimal string in the future.
+   * @see faker.number.hex(): For generating a random hexadecimal number in the future.
```
Comment (matthewmayer) [inline src/modules/datatype/index.ts]: i dont think the "in the future" is necessary

File: src/modules/finance/index.ts [hunk 13]
```diff
@@ -613,9 +613,9 @@ export class FinanceModule extends ModuleBase {
    * Returns a random currency object, containing `code`, `name `and `symbol` properties.
    *
    * @see
-   * faker.finance.currencyCode()
-   * faker.finance.currencyName()
-   * faker.finance.currencySymbol()
+   * faker.finance.currencyCode(): For generating specifically the currency code.
+   * faker.finance.currencyName(): For generating returns specifically the currency name.
+   * faker.finance.currencySymbol(): For generating returns specifically the currency symbol.
```
Comment (matthewmayer) [inline src/modules/finance/index.ts]: remove "returns" on both lines

File: src/modules/helpers/index.ts [hunk 14]
```diff
@@ -325,7 +325,7 @@ export class SimpleHelpersModule extends SimpleModuleBase {
    *
    * @param string The template string to parse. Defaults to `''`.
    *
-   * @see faker.helpers.fromRegExp()
+   * @see faker.helpers.fromRegExp(): For generating a string matching the given regex like expressions.
```
Comment (matthewmayer) [inline src/modules/helpers/index.ts]: hyphenate regex-like

File: src/modules/random/index.ts [hunk 15]
```diff
@@ -18,8 +18,8 @@ export class RandomModule extends ModuleBase {
   /**
    * Returns a random word.
    *
-   * @see faker.lorem.word()
-   * @see faker.word.sample()
+   * @see faker.lorem.word(): For generating a random placeholder word.
+   * @see faker.word.sample(): For generating a random existing word.
```
Comment (matthewmayer) [inline src/modules/random/index.ts]: perhaps "real" instead of "existing"

File: src/modules/random/index.ts [hunk 16]
```diff
@@ -157,8 +157,8 @@ export class RandomModule extends ModuleBase {
    * @param count.min The minimum number of words. Defaults to `1`.
    * @param count.max The maximum number of words. Defaults to `3`.
    *
-   * @see faker.lorem.words()
-   * @see faker.word.words()
+   * @see faker.lorem.words(): For generating a sequence of random placeholder words.
+   * @see faker.word.words(): For generating a sequence of random existing words.
```
Comment (matthewmayer) [inline src/modules/random/index.ts]: perhaps "real" instead of "existing"

File: src/modules/string/index.ts [hunk 17]
```diff
@@ -338,7 +338,7 @@ export class StringModule extends SimpleModuleBase {
    * @param options.length The number or range of characters to generate after the prefix. Defaults to `1`.
    * @param options.prefix Prefix for the generated number. Defaults to `'0b'`.
    *
-   * @see faker.number.binary() If you would like to generate a `binary number` (within a range).
+   * @see faker.number.binary(): For generating a `binary number` (within a range).
```
Comment (matthewmayer) [inline src/modules/string/index.ts]: are the backticks needed here?

**Review by matthewmayer — CHANGES_REQUESTED** [review]

File: src/modules/datatype/index.ts (hunk 10 above)
Comment (ST-DDT) [reply src/modules/datatype/index.ts]:
```suggestion
   * @see faker.number.int(): For generating a random integer.
```

File: src/modules/datatype/index.ts (hunk 11 above)
Comment (ST-DDT) [reply src/modules/datatype/index.ts]:
```suggestion
   * @see faker.number.float(): For generating a random floating-point number.
```

File: src/modules/datatype/index.ts (hunk 12 above)
Comment (ST-DDT) [reply src/modules/datatype/index.ts]:
```suggestion
   * @see faker.string.hexadecimal(): For generating a random hexadecimal string.
   * @see faker.number.hex(): For generating a random hexadecimal number.
```

File: src/modules/finance/index.ts (hunk 13 above)
Comment (ST-DDT) [reply src/modules/finance/index.ts]:
```suggestion
   * faker.finance.currencyName(): For generating specifically the currency name.
   * faker.finance.currencySymbol(): For generating specifically the currency symbol.
```

File: src/modules/helpers/index.ts (hunk 14 above)
Comment (ST-DDT) [reply src/modules/helpers/index.ts]:
```suggestion
   * @see faker.helpers.fromRegExp(): For generating a string matching the given regex-like expressions.
```

File: src/modules/random/index.ts (hunk 15 above)
Comment (ST-DDT) [reply src/modules/random/index.ts]:
```suggestion
   * @see faker.word.sample(): For generating a random real word.
```

File: src/modules/random/index.ts (hunk 16 above)
Comment (ST-DDT) [reply src/modules/random/index.ts]:
```suggestion
   * @see faker.word.words(): For generating a sequence of random real words.
```

File: src/modules/string/index.ts (hunk 17 above)
Comment (ST-DDT) [reply src/modules/string/index.ts]: They have been there already.

```suggestion
   * @see faker.number.binary(): For generating a binary number (within a range).
```

File: src/modules/string/index.ts [hunk 18]
```diff
@@ -390,7 +390,7 @@ export class StringModule extends SimpleModuleBase {
    * @param options.length The number or range of characters to generate after the prefix. Defaults to `1`.
    * @param options.prefix Prefix for the generated number. Defaults to `'0o'`.
    *
-   * @see faker.number.octal() If you would like to generate an `octal number` (within a range).
+   * @see faker.number.octal(): For generating an `octal number` (within a range).
```
Comment (ST-DDT) [inline src/modules/string/index.ts]:
```suggestion
   * @see faker.number.octal(): For generating an octal number (within a range).
```

File: src/modules/string/index.ts [hunk 19]
```diff
@@ -543,7 +543,7 @@ export class StringModule extends SimpleModuleBase {
    * @param options.allowLeadingZeros Whether leading zeros are allowed or not. Defaults to `true`.
    * @param options.exclude An array of digits which should be excluded in the generated string. Defaults to `[]`.
    *
-   * @see faker.number.int() If you would like to generate a `number` (within a range).
+   * @see faker.number.int(): For generating a `number` (within a range).
```
Comment (ST-DDT) [inline src/modules/string/index.ts]:
```suggestion
   * @see faker.number.int(): For generating a number (within a range).
```

**Review by matthewmayer — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**